### ISSUE TITLE
Some blocks don't have the general block name class `wp-block-woocommerce-{name}` which is needed for compatibility with core features

### ISSUE THREAD

**Issue by daviedR** [woothemes/woocommerce#49739]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Some of WooCommerce blocks don't have the general CSS class name `wp-block-woocommerce-{name}`. This class is needed for compatibility with WordPress core features, such as theme styling via `theme.json`.

Here are some of the blocks that don't have the `wp-block-woocommerce-{name}` class name:

- [x] - woocommerce/breadcrumbs
- [x] - woocommerce/catalog-sorting
- [x] - woocommerce/product-image
- [x] - woocommerce/product-button
- [x] - woocommerce/order-confirmation-status
- [x] - woocommerce/order-confirmation-summary
- [x] - woocommerce/order-confirmation-totals-wrapper
- [x] - woocommerce/order-confirmation-totals
- [x] - woocommerce/order-confirmation-downloads-wrapper
- [x] - woocommerce/order-confirmation-downloads
- [x] - woocommerce/order-confirmation-shipping-wrapper
- [x] - woocommerce/order-confirmation-shipping-address
- [x] - woocommerce/order-confirmation-billing-wrapper
- [x] - woocommerce/order-confirmation-billing-address
- [x] - woocommerce/order-confirmation-additional-fields-wrapper
- [x] - woocommerce/order-confirmation-additional-fields
- [x] - woocommerce/order-confirmation-additional-information

_** These are some blocks that I noticed, there might be some other blocks with a similar issue._

### Styling via `theme.json` doesn't work

When we add block styles via `theme.json`, WordPress will automatically generate the CSS using the general `wp-block-woocommerce-{name}` class as the selector. For example, if we add this snippet:

```
{
    "styles": {
        "blocks": {
            "woocommerce/breadcrumbs": {
                "color": {
                    "text": "red"
                }
            }
        }
    }
}
```

The generated CSS is:

```
:root :where(.wp-block-woocommerce-breadcrumbs){color: red;}
```

This won't affect the breadcrumb style, because it doesn't have the `wp-block-woocommerce-breadcrumbs` class.

### Styling via Site Editor's Global Styles doesn't work

The block styling options on Site Editor's Global Styles panel also don't work, because Global Styles works on top of the `theme.json` values and generates the same CSS selector.

### Expected behavior

All WooCommerce blocks should add the `wp-block-woocommerce-{name}` class in the rendered block container HTML.

### Actual behavior

Some WooCommerce blocks only add the `wc-block-{name}` class, while the `wp-block-woocommerce-{name}` class is missing.

### Steps to reproduce

1. Add the block styling for `woocommerce/breadcrumbs` block via `theme.json` (using the snippet above).
2. Save the changes, and see the frontend result.
3. The styling has no effect on the frontend.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.1.2
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.6
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.3.8
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 8.4.0
(SecureTransport) LibreSSL/3.3.6

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 24
page: 19
post: 6
product: 18
product_variation: 7
revision: 1
shop_coupon: 1
shop_order_placehold: 1
wp_font_face: 2
wp_font_family: 1
wp_global_styles: 7
wp_navigation: 1
wp_template: 5
wp_template_part: 2
wpcode: 4

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 9.1.2

### Inactive Plugins (23) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
Gutenberg: by Gutenberg Team – 18.8.0
WooPayments: by WooCommerce – 7.9.2
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.6

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 81 KB

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/ -  Contains the woocommerce/cart block
Checkout: #9 - /checkout/ -  Contains the woocommerce/checkout block
My account: #10 - /account/
Terms and conditions: #458 - /terms-and-conditions/

### Theme ###

Name: Twenty Twenty-Four
Version: 1.2
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /.../public/wp-content/plugins/woocommerce/templates/block-notices/error.php
/.../public/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/.../public/wp-content/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

- [ ] Enabled Features: activity-panels
- [ ] analytics
- [ ] product-block-editor
- [ ] coupons
- [ ] core-profiler
- [ ] customize-store
- [ ] customer-effort-score-tracks
- [ ] import-products-task
- [ ] experimental-fashion-sample-products
- [ ] shipping-smart-defaults
- [ ] shipping-setting-tour
- [ ] homescreen
- [ ] marketing
- [ ] mobile-app-banner
- [ ] navigation
- [ ] onboarding
- [ ] onboarding-tasks
- [ ] product-custom-fields
- [ ] remote-inbox-notifications
- [ ] remote-free-extensions
- [ ] payment-gateway-suggestions
- [ ] shipping-label-banner
- [ ] subscriptions
- [ ] store-alerts
- [ ] transient-notices
- [ ] woo-mobile-welcome
- [ ] wc-pay-promotion
- [ ] wc-pay-welcome-page
- [ ] launch-your-store

Disabled Features: experimental-blocks
minified-js
pattern-toolkit-full-composability
product-pre-publish-modal
printful
settings
async-product-editor-category-field
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-07-21 07:59:40 +07:00
Options: ✔
Notes: 79
Onboarding: completed

### Action Scheduler ###

Complete: 559
Oldest: 2024-06-20 01:52:42 +0000
Newest: 2024-07-20 16:22:49 +0000

Failed: 2
Oldest: 2023-12-28 08:04:25 +0000
Newest: 2024-07-09 05:08:31 +0000

Pending: 1
Oldest: 2024-07-21 16:12:19 +0000
Newest: 2024-07-21 16:12:19 +0000


### Status report information ###

Generated at: 2024-07-20 23:22:54 +07:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by mikejolley** [comment]
Thanks for the report. This does seem like a high priority for the backlog. @woocommerce/rubik will need to review checkout/order confirmation blocks. @woocommerce/woo-fse will need to check product blocks. We can split this issue up later if needed.

Thanks again

**Comment by mikejolley** [comment]
I'm tacking all cart and checkout blocks, and a few of the sitewide ones like breadcrumbs.

For the @woocommerce/woo-fse teams I found this list of blocks which may need classname adjustments in the format `wp-block-namespace-blockname`:

- product-image
- product-rating
- product-rating-counter
- product-rating-stars
- product-search
- add-to-cart-form
- catalog-sorting
- product-average-rating
- product-button
- product-collection-no-results

Once I've raised my PR fixing the checkout ones, I'll reassign this issue.

Edit: Found here https://github.com/woocommerce/woocommerce/pull/51380/files

**Comment by thealexandrelara** [comment]
I just created a PR (https://github.com/woocommerce/woocommerce/pull/51558) covering the following blocks:

- Add to Cart with Options
- Product Image
- Product Rating
- Product Rating Stars
- Product Rating Counter

The remaining blocks that are still missing the `wp-block-namespace-blockname` classname are:

- product-search
- catalog-sorting
- product-average-rating
- product-button
- product-collection-no-results

**Comment by louwie17** [comment]
I addressed the remaining items in this PR: https://github.com/woocommerce/woocommerce/pull/51617
In particular:

- catalog-sorting
- product-average-rating
- product-button

I skipped both `product-search` and `product-collection-no-results` because `product-search` is a variation block of the `core/search` block, so won't include the class. The product collection no results block already included the class name, see screenshot:
<img width="1391" alt="Screenshot 2024-09-23 at 2 21 33 PM" src="https://github.com/user-attachments/assets/50e654a6-1dee-4117-b1e4-e6a6fecc55ff">

**Comment by louwie17** [comment]
Closing this issue, as this has been addressed as part of #51380, #51558, and #51617.
